### PR TITLE
[Assistants] Add pagination

### PR DESCRIPTION
**Pull request by mishig25**
# Add pagination to assistants page

| Lightmode | Darkmode | Mobile|
|----------|----------|----------|
| Screenshot 2024-02-04 at 3 30 03 PM   | Screenshot 2024-02-04 at 3 30 07 PM   | Screenshot 2024-02-04 at 3 30 31 PM   |

### CODE REVIEW

File: src/routes/assistants/+page.server.ts [hunk 1]
```diff
@@ -3,20 +3,36 @@ import { ENABLE_ASSISTANTS } from "$env/static/private";
 import { collections } from "$lib/server/database.js";
 import type { Assistant } from "$lib/types/Assistant";
 import { redirect } from "@sveltejs/kit";
+import type { Filter } from "mongodb";
+
+const NUM_PER_PAGE = 10;
```
Comment (mishig25) [inline src/routes/assistants/+page.server.ts:8]: @gary149 change this number to what's appropriate

File: src/lib/components/PaginationArrow.svelte [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+	import CarbonCaretLeft from "~icons/carbon/caret-left";
+	import CarbonCaretRight from "~icons/carbon/caret-right";
+
+	export let direction: "next" | "previous";
+	export let onClick: (dir: "next" | "previous") => void = () => {};
+	export let isDisabled = false;
+</script>
+
+<button
```
Comment (gary149) [inline src/lib/components/PaginationArrow.svelte:10]: This should be a link `<a>` to prev/next page (vs button with click event)

File: src/lib/components/Pagination.svelte [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	export let numItemsPerPage: number;
+	export let numTotalItems: number;
+	export let onChange: (newPageIndex: number) => void = () => {};
+	export let pageIndex: number;
+
+	let pageLinks: PageLink[] = [];
+
+	const numTotalPages = Math.ceil(numTotalItems / numItemsPerPage);
+
+	$: {
```
Comment (gary149) [inline src/lib/components/Pagination.svelte:31]: This seems a bit complex 👀

File: src/lib/components/Pagination.svelte [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+{#if numTotalPages > 1}
+	<nav>
+		<ul
+			class="flex select-none items-center justify-between space-x-2 text-gray-700 sm:justify-center dark:text-gray-300 {classNames}"
+		>
+			<li>
+				<PaginationArrow
+					direction="previous"
+					isDisabled={pageIndex - 1 < 0}
+					onClick={handleClickDirection}
```
Comment (gary149) [inline src/lib/components/Pagination.svelte:93]: no need for javascript, next/before should be links

File: src/routes/assistants/+page.svelte [hunk 5]
```diff
@@ -24,6 +26,13 @@
 		}
 		goto(newUrl);
 	};
+
+	const onPaginationChange = (newPageIndex: number) => {
+		pageIndex = newPageIndex;
+		const newUrl = new URL($page.url);
+		newUrl.searchParams.set("p", newPageIndex.toString());
+		goto(newUrl);
```
Comment (gary149) [inline src/routes/assistants/+page.svelte:34]: you can remove that too

**Review by gary149 — APPROVED** [review]

File: src/lib/components/Pagination.svelte (hunk 3 above)
Comment (mishig25) [reply src/lib/components/Pagination.svelte:31]: copied from hub. I will see what should be done for simplification

Comment (mishig25) [reply src/lib/components/Pagination.svelte]: handled in https://github.com/huggingface/chat-ui/pull/772/commits/914dbd9a6023b2ff49d98748e8fbf2e4e5cbadd1

File: src/routes/assistants/+page.svelte (hunk 5 above)
Comment (mishig25) [reply src/routes/assistants/+page.svelte]: handled in https://github.com/huggingface/chat-ui/pull/772/commits/53ed09bc5c6a17d95dcd844deeba44fd4f7482c0

File: src/lib/components/Pagination.svelte (hunk 4 above)
Comment (mishig25) [reply src/lib/components/Pagination.svelte]: handled in https://github.com/huggingface/chat-ui/pull/772/commits/53ed09bc5c6a17d95dcd844deeba44fd4f7482c0

File: src/lib/components/PaginationArrow.svelte (hunk 2 above)
Comment (mishig25) [reply src/lib/components/PaginationArrow.svelte]: handled in https://github.com/huggingface/chat-ui/pull/772/commits/53ed09bc5c6a17d95dcd844deeba44fd4f7482c0

**Review by gary149 — APPROVED** [review]
Ok!

File: src/routes/assistants/+page.server.ts (hunk 1 above)
Comment (gary149) [inline src/routes/assistants/+page.server.ts:8]:
```suggestion
const NUM_PER_PAGE = 24;
```

**Pull request merged**